test/unit/extension/lib/shopgate/CartExtensionPipelineTest.js

```diff
@@ -76,4 +77,23 @@ describe('CartExtensionPipeline - unit', () => {
 
     return subjectUnderTest.getCheckoutUrl().should.eventually.equal(expectedUrl)
   })
+  it('should return true when update product runs without error', function () {
+    bigCommerceCartRepositoryMock.expects('updateItems').once().returns()
+
+    return subjectUnderTest.updateProducts([{ productId: '1', quantity: 1 }]).should.eventually.equal(true)
+  })
+
+  it('should return false when update product encounters a non-breaking error', function () {
```

Comment:
You should add mock for piepline context and check that the warning is logged.

Reply:
created a spy for the error method of context.log. The test asserts that it is never called.